src/table.rs

```diff
@@ -41,6 +42,26 @@ pub struct Table {
 }
 
 impl Table {
+    /// Creates a new table from a schema and a vector of record batches.
+    ///
+    /// Returns an error if a record batch's schema fields do not match the
+    /// top-level schema's fields.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use arrow_array::RecordBatch;
+    /// use arrow_schema::{Schema, SchemaRef};
+    /// use geoarrow::{GeometryArrayTrait, array::PointArray, table::Table};
+    ///
+    /// let point = geo::point!(x: 1., y: 2.);
+    /// let array: PointArray<2> = vec![point].as_slice().into();
+    /// let field = array.extension_field();
+    /// let schema: SchemaRef = Schema::new(vec![field]).into();
+    /// let columns = vec![array.into_array_ref()];
+    /// let batch = RecordBatch::try_new(schema.clone(), columns).unwrap();
+    /// let table = Table::try_new(schema, vec![batch]).unwrap();
+    /// ```
     pub fn try_new(schema: SchemaRef, batches: Vec<RecordBatch>) -> Result<Self> {
```

Comment:
I've perennially been unsure whether arguments should be ordered `schema, batches` or `batches, schema`. Now in pyo3-arrow [I've consolidated on `batches, schema`](https://github.com/kylebarron/arro3/blob/16210f5893e0e7df0fc906aac90732d302c9a9a7/pyo3-arrow/src/table.rs#L39), so I have a slight preference to change this here to that ordering

Reply:
Yup classic! When I was going through this I found [pyarrow.RecordBatchReader.from_batches](https://arrow.apache.org/docs/python/generated/pyarrow.RecordBatchReader.html#pyarrow.RecordBatchReader.from_batches) which goes `schema, batches`. I'm +/- 0 on order, happy to implement whichever you prefer.

Reply:
🫠

Reply:
arrow-rs's `RecordBatchIterator` uses `batches, schema` https://docs.rs/arrow/latest/arrow/array/struct.RecordBatchIterator.html#method.new

Reply:
I suppose I'd lean towards keeping it consistent with arrow-rs there?

Reply:
Ok, I've updated everything in **geoarrow-rs** to be `batches, schema`: 5cd0e76156951aeeb67ddc23ad9618017f433c51

NB that [arrow-wasm](https://github.com/kylebarron/arrow-wasm) should probably switch to `batches, schema`, too, for consistency's sake.